pkg/injector/bootstrap_test.go

```diff
@@ -312,3 +325,201 @@ var _ = Describe("Test functions creating Envoy bootstrap configuration", func()
 		})
 	})
 })
+
+func TestListBootstrapSecrets(t *testing.T) {
+	testCases := []struct {
+		name       string
+		secrets    []*corev1.Secret
+		expSecrets []*corev1.Secret
+	}{
+		{
+			name: "get bootstrap secrets from k8s secrets",
+			secrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "notBootstrapSecret",
+						Namespace: "testNamespace",
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      bootstrapSecretPrefix + "proxyUUID",
+						Namespace: "testNamespace",
+					},
+				},
+			},
+			expSecrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      bootstrapSecretPrefix + "proxyUUID",
+						Namespace: "testNamespace",
+					},
+				},
+			},
+		},
+		{
+			name: "no bootstrap secrets in k8s secrets",
+			secrets: []*corev1.Secret{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "notBootstrapSecret",
+						Namespace: "testNamespace",
+					},
+				},
+			},
+			expSecrets: []*corev1.Secret{},
+		},
+	}
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("Running test case %d: %s", i, tc.name), func(t *testing.T) {
+			assert := tassert.New(t)
+
+			mockController := k8s.NewMockController(gomock.NewController(t))
+			mockController.EXPECT().ListSecrets().Return(tc.secrets)
+
+			certManager, err := certificate.FakeCertManager()
+			assert.Nil(err)
+
+			b := NewBootstrapSecretRotator(mockController, certManager, time.Duration(1))
+
+			actual := b.listBootstrapSecrets()
+			assert.ElementsMatch(tc.expSecrets, actual)
+		})
+	}
+}
+
+func TestRotateBootstrapSecrets(t *testing.T) {
+	assert := tassert.New(t)
+
+	testNs := "testNamespace"
+	proxyUUID := uuid.New()
+	secretName := bootstrapSecretPrefix + proxyUUID.String()
+
+	testCases := []struct {
+		name     string
+		certName string
+		secrets  []*corev1.Secret
+	}{
+		{
+			name:     "update bootstrap secret with new cert",
```

Comment:
should we have a test case to verify it doesn't rotate?

Reply:
and where there are more than one secret, etc?

Reply:
@jsturtevant not sure how we'd go about testing if it doesn't rotate, since that's determined by ShouldRotate(), any ideas?

Reply:
That would create a cert that's not about to expire, but `shouldRotate `also checks if the `signingIssuer `or `validatingIssuer` are different [here ](https://github.com/openservicemesh/osm/blob/main/pkg/certificate/manager.go#L200) and I don't think there's a way to set those variables unless I exported them

Reply:
@jsturtevant that does create a cert that's not going to expire, but `shouldRotate` also [checks if the `signingIssuer` or `validatingIssuer` are different](https://github.com/openservicemesh/osm/blob/main/pkg/certificate/manager.go#L200), and I don't think I can set those variables unless I export them 🤔

Reply:
I ended up setting the `signingIssuer` and `validatingIssuer` IDs to empty strings in `pkg/certificate/fake_manager.go`